subprojects/launcher/src/integTest/groovy/org/gradle/launcher/CommandLineIntegrationSpec.groovy

```diff
@@ -239,4 +239,30 @@ class CommandLineIntegrationSpec extends AbstractIntegrationSpec {
         6                      | ['--parallel', '--max-workers=6', '-Dorg.gradle.workers.max=4']
         4                      | ['-Dorg.gradle.parallel=true', '--max-workers=4']
     }
+
+    @Unroll
+    def "Set log level using org.gradle.log.level"() {
```

Comment:
We should add some more coverage to verify that the usual locations are supported, and that they are applied in the correct order of precedence. For example, that a log level command-line option wins over a `-D` command-line option, which wins over `GRADLE_OPTS=-D`, which wins over ~/gradle/gradle.properties, etc.